process/roundActivation/roundActivation.go

```diff
@@ -3,50 +3,31 @@ package roundActivation
 import (
 	"reflect"
 
-	"github.com/ElrondNetwork/elrond-go-core/core/check"
 	"github.com/ElrondNetwork/elrond-go/config"
 	"github.com/ElrondNetwork/elrond-go/process"
 )
 
 type roundActivation struct {
-	roundHandler   process.RoundHandler
 	roundByNameMap map[string]uint64
 }
 
 // NewRoundActivation creates a new round activation handler component
-func NewRoundActivation(roundHandler process.RoundHandler, config config.RoundConfig) (process.RoundActivationHandler, error) {
-	if check.IfNil(roundHandler) {
-		return nil, process.ErrNilRoundHandler
-	}
-
+func NewRoundActivation(config config.RoundConfig) (process.RoundActivationHandler, error) {
 	configMap, err := getRoundsByNameMap(config)
 	if err != nil {
 		return nil, err
 	}
 
 	return &roundActivation{
-		roundHandler:   roundHandler,
 		roundByNameMap: configMap,
 	}, nil
 }
 
 // IsEnabled checks if the queried round flag name is enabled in the queried round
 func (ra *roundActivation) IsEnabled(name string, round uint64) bool {
-	if r, exists := ra.roundByNameMap[name]; exists {
-		return r == round && ra.isCurrentRound(round)
-	}
-
-	return false
-}
-
-func (ra *roundActivation) isCurrentRound(round uint64) bool {
-	return round == uint64(ra.roundHandler.Index())
-}
-
-// IsEnabledInCurrentRound checks if the queried round flag name is enabled in current round
-func (ra *roundActivation) IsEnabledInCurrentRound(name string) bool {
-	if round, exists := ra.roundByNameMap[name]; exists {
-		return ra.isCurrentRound(round)
+	r, exists := ra.roundByNameMap[name]
```

Comment:
no mutex protection? The method is exposed and this struct is prone to be used in concurrent manner.

Reply:
I know it does only reads but the code is fragile in this way. 
Reason: adding a write **with mutex protection** in a new function but forgetting to add the mutex in all other methods could easily panic the node. And we might encounter that in production environment.

Reply:
Added mutex